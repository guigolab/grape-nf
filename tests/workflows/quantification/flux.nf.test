nextflow_workflow {

    name "Test Workflow quantification"
    script "workflows/quantification/flux.nf"
    workflow "quantification"

    tag "workflow"

    test("Should run quantification with Flux Capacitor") {

        when {
            params {
                stepList = [ "quantification" ]
            }
            workflow {
                """
                input[0] = file("${baseDir}/data/genome.fa")
                input[1] = file("${baseDir}/data/annotation.gtf")
                input[2] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true, "MATE2_SENSE" ]
                ])
                input[3] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toTranscriptome.bam"), "bam", "TranscriptomeAlignments", true, "MATE2_SENSE" ]
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
