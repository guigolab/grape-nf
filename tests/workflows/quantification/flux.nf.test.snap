{
    "Should run quantification with Flux Capacitor": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.gff:md5,74f9de8e31ff2c24bc45af5f8ec73530",
                        "gtf",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.gtf:md5,1a039c489fdeb3e0ab4ff768424b5997",
                        "gtf",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "genes": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.gff:md5,74f9de8e31ff2c24bc45af5f8ec73530",
                        "gtf",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "isoforms": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.gtf:md5,1a039c489fdeb3e0ab4ff768424b5997",
                        "gtf",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T17:54:38.170082"
    }
}